zboxcore/sdk/playlist_consensus.go

```diff
@@ -0,0 +1,116 @@
+package sdk
+
+import (
+	"encoding/json"
+	"sort"
+)
+
+type playlistConsensus struct {
+	files       map[string]PlaylistFile
+	consensuses map[string]*Consensus
+
+	threshConsensus float32
+	fullConsensus   float32
+	consensusOK     float32
+}
+
+func createPlaylistConsensus(fullConsensus, threshConsensus, consensusOK float32) *playlistConsensus {
+	return &playlistConsensus{
+		files:           make(map[string]PlaylistFile),
+		consensuses:     make(map[string]*Consensus),
+		threshConsensus: threshConsensus,
+		fullConsensus:   fullConsensus,
+		consensusOK:     consensusOK,
+	}
+}
+
+func (c *playlistConsensus) AddFile(body []byte) error {
+	file := PlaylistFile{}
+
+	if err := json.Unmarshal([]byte(body), &file); err != nil {
+		return err
+	}
+
+	_, ok := c.files[file.LookupHash]
+
+	if ok {
+		c.consensuses[file.LookupHash].Done()
+	} else {
+		cons := &Consensus{}
+
+		cons.Init(c.threshConsensus, c.fullConsensus, c.consensusOK)
+		cons.Done()
+
+		c.consensuses[file.LookupHash] = cons
+		c.files[file.LookupHash] = file
+	}
+
+	return nil
+}
+
+func (c *playlistConsensus) Add(body []byte) error {
```

Comment:
Rename to `AddFiles()` perhaps would make it easier for user to get what's the difference between it and the AddFile above.